python/documentation/create_operations_doc.py

```diff
@@ -0,0 +1,71 @@
+''' Script for generating Python documentation from the ASCIIDOC documentation.
+
+
+
+Usage: python create_operations_doc.py
+'''
+import glob
+import os
+import re
+import textwrap
+
+from lynx.kite import _python_name
+
+
+FILES_TO_EXCLUDE = ['glossary.asciidoc', 'index.asciidoc']
+ASCIIDOC_OPERATIONS_PATH = '../../web/app/help/operations/'
+OUTPUT_PATH = '../remote_api/src/lynx/operations.py'
+HEADER = f'''\'\'\' Python documentation for the operations in Lynxkite.
+This document has been automatically generated.
+\'\'\'
+'''
+
+BODY_REGEX = re.compile(r'^$.*(?P<body>[^=]*)', re.MULTILINE)
+PARAMS_REGEX = re.compile(r'\[p-(?P<attr>.*)\].*\n(?P<desc>[^=|\[]*)', re.MULTILINE)
+
+
+def load_file(path):
+  with open(path, 'r') as fh:
+    return fh.read()
+
+
+def generate_function(operation_name, path):
+  ''' Generates the documentation for a given operation. '''
+  content = load_file(path)
+
+  body = textwrap.indent(re.search(BODY_REGEX, content).groups()[0].strip(), '  ')
+  params = re.findall(PARAMS_REGEX, content)
+  params = [] if not params else params
+
+  params_text = ''
+  for name, desc in params:
+    params_text += textwrap.indent(f':param {name}: {textwrap.indent(desc.strip(), "  ")}\n', '  ')
+  params_text = params_text.rstrip()
+  return f'''
+def {operation_name}(*args, **kwargs):
```

Comment:
Since you already know the parameter names, maybe it would not be much more trouble to put them in the argument list. If it would allow autocompletion for the parameter names when used in a notebook that would be super convenient.

Reply:
I was considering this myself as well. However, the problem that I see is that some parameters are `optional` (with a default value) whereas others are `required`. I'm not always able to infer this from the asciidocs. Unless I'm wrong here, do you think it's worth making the tradeoff of adding the parameters, even thought it comes at the cost of correctness?

Reply:
I don't think you can figure out the default values, so it would always appear as required, right? But what exactly would be the effect?
- If the documentation says `pageRank(weight, iterations)`, that's clearly better than if it says `pageRank(*args, **kwargs)`. The text for the parameter may clarify whether there is a default.
- If I type `pageRank(wei<TAB>` in a notebook and it autocompletes that's fantastic.

Would there be any downside, do you think?

Reply:
I'm convinced. I'll implement it.

Reply:
Done.